src/main/java/org/sopt/diary/service/DiaryService.java

```diff
@@ -0,0 +1,133 @@
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DiaryService {
+
+	private final DiaryRepository diaryRepository;
+	private final DiaryRateLimiter diaryRateLimiter;
+	private static final int DEFAULT_PAGE_SIZE = 10;
+	private static final int DEFAULT_PAGE_NUMBER = 0;
+
+	// bean 주입 di
+	public DiaryService(DiaryRepository diaryRepository, DiaryRateLimiter diaryRateLimiter) {
+		this.diaryRepository = diaryRepository;
+		this.diaryRateLimiter = diaryRateLimiter;
+	}
+
+	// 일기 작성
+	public Diary createDiary(DiaryCreateRequest diaryCreateRequest) {
+		if(!diaryRateLimiter.tryConsume()){
+			throw new TooManyRequestsException(DiaryErrorCode.DIARY_CREATE_TIME_LIMIT_LEFT);
+		}
+
+		boolean existsByTitle = diaryRepository.existsByTitle(diaryCreateRequest.title());
+		if (existsByTitle) {
+			throw new ConflictException(DiaryErrorCode.DIARY_TITLE_ALREADY_EXISTS);
+		}
+
+		final DiaryEntity diaryEntity = new DiaryEntity(diaryCreateRequest.title(), diaryCreateRequest.body(),
+			diaryCreateRequest.category());
+		diaryRepository.save(diaryEntity);
+
+		final Diary diary = convertToDiary(diaryEntity);
+
+		return diary;
+	}
+
+	// 일기 상세 조회
+	public Diary getDiaryDetail(DiaryDetailRequest diaryDetailRequest) {
+		final DiaryEntity diaryEntity = diaryRepository.findById(diaryDetailRequest.id())
+			.orElseThrow(() -> new NotFoundException(DiaryErrorCode.DIARY_NOT_FOUND));
+		final Diary diary = convertToDiary(diaryEntity);
+
+		return diary;
+	}
+
+	// db에서 가져온 diaryEntity를 diary 객체로 만들어서 list로 리턴
+	// 일기 목록 조회
+	public List<Diary> getList(Category category, OrderBy orderBy) {
+		Pageable pageable;
+
+		if (orderBy == OrderBy.BODY_LENGTH) {
+			pageable = PageRequest.of(DEFAULT_PAGE_NUMBER, DEFAULT_PAGE_SIZE);
+
+			Page<DiaryEntity> diaryEntityPage;
+			if (category == null) {
+				diaryEntityPage = diaryRepository.findAllOrderByBodyLengthDesc(pageable);
+			} else {
+				diaryEntityPage = diaryRepository.findByCategoryOrderByBodyLengthDesc(category, pageable);
+			}
+			return diaryEntityPage.getContent().stream().map(this::convertToDiary).collect(Collectors.toList());
+		} else {
+			Sort sort = Sort.by(Sort.Direction.DESC, "createdAt");
+			pageable = PageRequest.of(DEFAULT_PAGE_NUMBER, DEFAULT_PAGE_SIZE, sort);
+
+			Page<DiaryEntity> diaryEntityPage;
+			if (category == null) {
+				diaryEntityPage = diaryRepository.findAll(pageable);
+			} else {
+				diaryEntityPage = diaryRepository.findByCategory(category, pageable);
+			}
+			return diaryEntityPage.getContent().stream().map(this::convertToDiary).collect(Collectors.toList());
+		}
+	}
+
+	// 일기 수정
+	public Diary updateDiary(DiaryUpdateRequest diaryUpdateRequest) {
+		final DiaryEntity diaryEntity = diaryRepository.findById(diaryUpdateRequest.id())
+			.orElseThrow(() -> new NotFoundException(DiaryErrorCode.DIARY_NOT_FOUND));
+
+		if (!diaryEntity.getTitle().equals(diaryUpdateRequest.title())){
```

Comment:
이 조건은 제거되어도 될것같은데 어떻게 생각하실까요?
```suggestion
		if (!diaryEntity.getTitle().equals(diaryUpdateRequest.title())){
```

Reply:
해당 조건을 제거하면 제목을 변경하지 않고 다른 필드만 수정하려고 할 때도 제목이 이미 존재한다고 판단하여 에러가 발생하게 돼서 포함했습니다!